File: distribute_candy.js
module.exports = { 
    //param A : array of integers
    //return an integer
       candy : function(A){
           let candy = [];
   
           for (let i=0;i<A.length;i++){
               candy.push(1);
           }
           for (let i=1; i<=A.length;i++){
               if(A[i]>A[i-1]){
                   candy[i] = candy[i-1] + 1;
               }
           }
           for (let i=A.length-2;i>=0;i--){
               if(A[i]>A[i+1]){
                   candy[i] = Math.max(candy[i],candy[i+1]+1);
               }
           }
   
           let total_candies = 0;
           for (let i=0;i<candy.length;i++){
               total_candies += candy[i];
           }
           return total_candies;
       }
   };
   

File: distribute_candy.py
class Solution:
    # @param A : list of integers
    # @return an integer
    def candy(self, A):
        A_candies = [1]*len(A)
        for i in range(1,len(A)):
            if A[i-1] < A[i]:
                A_candies[i] = A_candies[i-1]+1
        
        for j in range(len(A)-2,-1,-1):
            if A[j] > A[j+1]:
                A_candies[j] = max(A_candies[j],A_candies[j+1]+1) 
        s = sum(A_candies)
        return s